### ISSUE TITLE
Warning if levels of indentation exceeds, say, 4

### ISSUE THREAD

**Issue by tbodt** [rust-lang/rust-clippy#2169]
I'd like to have a warning from Clippy if I write code with more than 4 levels of indentation, to incentivize me to split up complicated functions into less complicated ones.

**Comment by Manishearth** [comment]
Clippy already warns for complicated functions, base on cyclomatic complexity. The threshold is configurable.

**Comment by Centri3** [comment]
I agree with closing this, but for a different reason: `excessive_nesting` has been added way after 2017. This specifically looks for nesting so isn't really based on complexity or formatting (except in the case of blocks with only an expression)
